migrations/versions/d25d457860de_lowercase_discount_code.py

```diff
@@ -0,0 +1,34 @@
+"""lowercase discount code
+
+Revision ID: d25d457860de
+Revises: 6f7b6fad3f54
+Create Date: 2019-05-19 02:06:14.730180
+
+"""
+
+from alembic import op
+import sqlalchemy as sa
+import sqlalchemy_utils
+
+
+# revision identifiers, used by Alembic.
+revision = 'd25d457860de'
+down_revision = '6f7b6fad3f54'
+
+
+def upgrade():
+    op.execute("UPDATE discount_codes SET code = concat(code, '_') where code not in (SELECT DISTINCT ON (upper(code)) code FROM discount_codes group by event_id, code);",
+               execution_options=None)
+    op.execute("alter table discount_codes alter column code type citext;",
+               execution_options=None)
+    op.execute("alter table discount_codes add constraint uq_discount_codes unique(event_id, code);",
+               execution_options=None)
+
+
+def downgrade():
+    op.execute("UPDATE discount_codes SET code = left(code, length(code)-1) where right(code, 1) = '_';",
+               execution_options=None)
+    op.execute("alter table discount_codes alter column code type text;",
+               execution_options=None)
+    op.execute("alter table discount_codes constraint uq_discount_codes;",
```

Comment:
This query is wrong. Please verify everything before pushing

Reply:
Such a stupid mistake. Forgot to write drop. Sorry. Will update. Any idea why the build is failing?